src/autopas/LogicHandler.h

```diff
@@ -1077,88 +1206,326 @@ void LogicHandler<Particle>::remainderHelperBufferHaloBuffer(
 }
 
 template <typename Particle>
-template <class Functor>
-std::tuple<Configuration, std::unique_ptr<TraversalInterface>, bool> LogicHandler<Particle>::selectConfiguration(
-    Functor &functor) {
+template <class TriwiseFunctor>
+typename LogicHandler<Particle>::IterationMeasurements LogicHandler<Particle>::iterateTriwise(
+    TriwiseFunctor &functor, TraversalInterface<InteractionTypeOption::threeBody> &traversal) {
+  const bool doListRebuild = not neighborListsAreValid();
+  const auto &configuration = _autoTuner3B->getCurrentConfig();
+  auto &container = _containerSelector.getCurrentContainer();
+
+  autopas::utils::Timer timerTotal;
+  autopas::utils::Timer timerRebuild;
+  autopas::utils::Timer timerIterateTriwise;
+  autopas::utils::Timer timerRemainderTraversal;
+
+  const bool energyMeasurementsPossible = _autoTuner3B->resetEnergy();
+  timerTotal.start();
+
+  functor.initTraversal();
+  //   TODO: Add list rebuilds for 3-Body
+  //    if (doListRebuild) {
+  //      timerRebuild.start();
+  //      container.rebuildNeighborLists(&traversal);
+  //      timerRebuild.stop();
+  //    }
```

Comment:
Because it was not yet needed and caused problems :D
Containers now have both overloaded methods for given pairwise and triwise traversals:
```cpp
void rebuildNeighborLists(PairwiseTraversal *traversal) {}
void rebuildNeighborLists(TriwiseTraversal *traversal) {}
```
Implemented in dbfbe32fa2f92b9d10e723866a735f4a061c0012